### DISCUSSION TITLE
How to create task with group managed service account (gMSA) account?

### DISCUSSION THREAD

**Discussion by the-mike-anderson** [dahall/TaskScheduler#996] [Q&A]
Hi all,

I've just started converting to a .NET 8.0 MVC web application from a .NET 4.8 console application using the TaskScheduler COM library.  Using that library, I am able to create scheduled tasks with a gMSA account by setting the password to null.  Using this library, I am unable to create the task with a gMSA account.  I get the following error:

`The task has been configured with an unsupported combination of account settings and run time options. (0x80041314)`

I am setting the `forceV1` parameter to true in this line of code:

`SchedTask.TaskService ts = new SchedTask.TaskService(options?.ServerWithBackslashes, (string)null, (string)null, (string)null, true);`

I am unable to connect to Windows 2019 server otherwise.  The failure occurs on RegisterTaskDefinition call below:

```
using SchedTask = Microsoft.Win32.TaskScheduler;

...

ts.RootFolder.RegisterTaskDefinition(
					taskname,
					td,
					SchedTask.TaskCreation.CreateOrUpdate,
					UserName,         // is of the pattern domain\gmsa_account$
					UserPassword,     // i've tried null and empty string
					logonType,        // set to Microsoft.Win32.TaskScheduler.TaskLogonType.Password
					null
					);
```

I'd like to add that when I use a normal account name and password, the scheduled task is created just fine.

**Comment by dahall** [comment]
Do not set forceV1 = true; that pushes it back to WinXP settings. Let me know if that fixes things.

**Reply by the-mike-anderson** [reply]
Thank you for checking in David.  I had it set to forceV1 as a hail Mary approach to getting it to work.  I have had no success with it set or not set.  Also, I should have gotten back here and closed out the question.  I've already moved on with another solution where I build a dynamic PowerShell script that gets remotely executed against the target machine.
